### PR TITLE
added saltstack last version which compiles and runs on OSX Yosemite

### PR DESCRIPTION
**Pull request by julienlavergne**

### CODE REVIEW

File: saltstack.rb [hunk 1]
```diff
@@ -0,0 +1,95 @@
+class Saltstack < Formula
+
```
Comment (DomT4) [inline saltstack.rb]: No need for the whitespacing up here.

File: saltstack.rb [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  homepage "http://www.saltstack.org"
+
+  url "https://github.com/saltstack/salt/archive/v2015.2.0rc2.tar.gz"
+
+  sha256 'be71c1f2f9f878d5f958396620983c5981f55eaf32913e7f28c129c35f37657a'
```
Comment (DomT4) [inline saltstack.rb]: Double quotes everywhere.

File: saltstack.rb [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+  bottle do
```
Comment (DomT4) [inline saltstack.rb]: Kill the bottle block.

File: saltstack.rb [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    sha1 "4ef3922ffd2b36d775f22fce055ebf692d1e14b7" => :yosemite
+    sha1 "fbb96fe311befd1f68c063c398bfb4a011fc2dc4" => :mavericks
+    sha1 "738d8efecb7b916f10acf8707822945623226524" => :mountain_lion
+  end
+
+  head "https://github.com/saltstack/salt.git", :branch => "develop", :shallow => false
```
Comment (DomT4) [inline saltstack.rb]: Remove the ` head `.

File: saltstack.rb [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+  depends_on :python if MacOS.version <= :snow_leopard
+  depends_on "swig304" => :build
```
Comment (DomT4) [inline saltstack.rb]: Please make sure the failure to build with current ` swig ` has been reported upstream and include that bug report link in the formula.

File: saltstack.rb [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+  depends_on "zeromq"
+  depends_on "libyaml"
+
+  # Don't depend on Homebrew's openssl due to upstream build issues with non-system OpenSSL in M2Crypto
+  # See: https://github.com/martinpaljak/M2Crypto/issues/11
+
+  resource "m2crypto" do
+    url "https://pypi.python.org/packages/source/M/M2Crypto/M2Crypto-0.22.3.tar.gz"
+    sha1 "c5e39d928aff7a47e6d82624210a7a31b8220a50"
```
Comment (DomT4) [inline saltstack.rb]: Bump all of these to ` sha256 `.

File: saltstack.rb [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+  resource "requests" do
+    url "https://pypi.python.org/packages/source/r/requests/requests-2.3.0.tar.gz"
+    sha1 "f57bc125d35ec01a81afe89f97dc75913a927e65"
+  end
+
+  def install
+    ENV["PYTHONPATH"] = lib+"python2.7/site-packages"
+    ENV.prepend_create_path "PYTHONPATH", libexec+"lib/python2.7/site-packages"
+
+    resources.each do |r|
```
Comment (DomT4) [inline saltstack.rb]: Try to modernise the Python handling similar to [jrnl upstream](https://github.com/Homebrew/homebrew/blob/master/Library/Formula/jrnl.rb#L53-L65).

**Comment by DomT4** [comment]
You'll need to rename the formula and add a ` conflicts_with ` to avoid bumping into Homebrew/homebrew's Saltstack formula.

**Comment by julienlavergne** [comment]
It’s done 

thank you for yours advices



-- 
Julien Lavergne
Sent with Airmail

On April 13, 2015 at 9:54:21 PM, Dominyk Tiller (notifications@github.com) wrote:

You'll need to rename the formula and add a conflicts_with to avoid bumping into Homebrew/homebrew's Saltstack formula.

**Comment by DomT4** [comment]
You seem to have removed the file completely? ` Files changed 0 `

**Comment by mistydemeo** [comment]
If this is a release candidate, this is probably better suited to being a `devel do` block in the main saltstack formula.

**Comment by DomT4** [comment]
Agree with Misty. My main concern with that was the `swig` dep being turned into a `swig304` dep, and I don't know how the core feels about build-time tap non-optional deps.

**Comment by julienlavergne** [comment]
just added a commit with a new name Formula name

so just rename with -dev suffix?

-- 
Julien Lavergne
Sent with Airmail

On April 13, 2015 at 10:11:15 PM, Misty De Meo (notifications@github.com) wrote:

If this is a release candidate, this is probably better suited to being a devel do block in the main saltstack formula.

**Comment by julienlavergne** [comment]
Suffix -latest is according to IMHO as dev version

-- 
Julien Lavergne
Sent with Airmail

On April 13, 2015 at 10:12:30 PM, Julien Lavergne (lavergne.julien@gmail.com) wrote:

just added a commit with a new name Formula name

so just rename with -dev suffix?

-- 
Julien Lavergne
Sent with Airmail

On April 13, 2015 at 10:11:15 PM, Misty De Meo (notifications@github.com) wrote:

If this is a release candidate, this is probably better suited to being a devel do block in the main saltstack formula.

**Comment by julienlavergne** [comment]
Saltstack on OSX has a big dependency problem;

To maintain saltstack last version compilation and running under OSX yosemite swig304 is necessary.

swig maintainers are busy to correctly release bug fixes for saltstack.  

-- 
Julien Lavergne
Sent with Airmail

On April 13, 2015 at 10:12:35 PM, Dominyk Tiller (notifications@github.com) wrote:

Agree with Misty. My main concern with that was the swig dep being turned into a swig304 dep, and I don't know how the core feels about build-time tap non-optional deps.

**Comment by DomT4** [comment]
> so just rename with -dev suffix?

It would be the actual version or branch, usually. Something like ` saltstack201520 `. That's not ideal though, given that they may do future devel releases and that name would no longer be true.

Provisionally I agree with @mistydemeo here, but will defer whether the ` swig304 ` dep is problematic for the core to Misty. ` devel ` versions can have looser rules in places, but given I'm not a maintainer at Homebrew/homebrew I don't want to make that call for them.

**Comment by julienlavergne** [comment]
Homebrew doesn't want this formula because it's a dev release

**Comment by DomT4** [comment]
I don't think that's the position of @Homebrew/owners. From Jack, Tim & Misty's comments here and there it looks like the debate primarily is *"Can we include this given the swig problems"* and *"Is this more of a* `devel` *block-worthy release than a versions release*"?

I think it's worth keeping these discussions up in Homebrew/homebrew for now, and making a decision there. Having two discussions across two PRs can be hard to keep track of :smile_cat:.

**Comment by julienlavergne** [comment]
I understand.

The dependency with swig304 is problematic.  

-- 
Julien Lavergne
Sent with Airmail

On April 14, 2015 at 6:11:34 PM, Dominyk Tiller (notifications@github.com) wrote:

I don't think that's the position of @Homebrew/owners. From Jack, Tim & Misty's comments here and there it looks like the debate primarily is "Can we include this given the swig problems" and "Is this more of a devel block-worthy release than a versions release"?

I think it's worth keeping these discussions up in Homebrew/homebrew for now, and making a decision there. Having two discussions across two PRs can be hard to keep track of .

**Comment by DomT4** [comment]
https://github.com/Homebrew/homebrew/pull/38783

**Comment by DomT4** [comment]
@BrewTestBot You were premature :wink:. I'll reopen this if it proves necessary later.

**Pull request closed without merging**